testcases/kernel/syscalls/pwrite/pwrite02.c

```diff
@@ -198,17 +172,19 @@ static void test_efault(void)
 	int fd;
 	char *buf = sbrk(0);
 
-	fd = SAFE_OPEN(cleanup, TEMPFILE, O_RDWR | O_CREAT, 0666);
+	fd = SAFE_OPEN(TEMPFILE, O_RDWR | O_CREAT, 0666);
 
 	TEST(pwrite(fd, buf, K1, 0));
 
-	print_test_result(errno, EFAULT);
+	print_test_result(TEST_ERRNO, EFAULT);
 
-	SAFE_CLOSE(cleanup, fd);
+	SAFE_CLOSE(fd);
 }
 #endif
 
-static void cleanup(void)
-{
-	tst_rmdir();
-}
+static struct tst_test test = {
+	.needs_tmpdir = 1,
+	.setup = setup,
+	.test = verify_pwrite,
+	.tcnt = ARRAY_SIZE(testfunc),
+};
```

Comment:
Well we usually tend to use more data driven approach for testcases that check for the invalid syscall inputs, just have a look at syscalls/read/read02.c or do 'git grep struct tcase' but if you do not want to spend more time on this the code is clean enough to be accepted as well.